.github/workflows/Deploy.yaml

```diff
@@ -97,19 +97,28 @@ jobs:
           $errorActionPreference = "Stop"
           $end2endRuns = (gh api "/repos/$($env:GITHUB_REPOSITORY)/actions/workflows/E2E.yaml/runs?per_page=100&branch=main" | ConvertFrom-Json).workflow_runs
           $latestSha = (gh api /repos/$($env:GITHUB_REPOSITORY)/commits/main | ConvertFrom-Json).sha
-          $latestRun = $end2endruns | Where-Object { $_.Name -eq "End to end tests - $latestSha" } | select-object -first 1
+          $latestRun = $end2endRuns | Where-Object { $_.Name -eq "End to end tests - $latestSha" } | select-object -first 1
+
           if (!$latestRun) {
-            throw "No End to end tests run found for the latest commit on main"
+            throw "No End to end tests run found for the latest commit on main (sha: $latestSha)"
           }
+
+          Write-Host "Latest E2E test run on main (sha: $latestSha): $($latestRun.html_url)"
           if ($latestRun.status -ne 'completed') {
             throw "End to end tests run for the latest commit on main is not completed - see $($latestRun.html_url)"
           }
           if ($latestRun.conclusion -ne 'success') {
             throw "End to end tests run for the latest commit on main did not succeed - see $($latestRun.html_url)"
           }
           $allJobs = (gh api --paginate /repos/$($env:GITHUB_REPOSITORY)/actions/runs/$($latestRun.id)/jobs | ConvertFrom-Json).jobs
+
+          $nonCriticalJobs = @("RemoveRepositories")
```

Comment:
Maybe we should run this workflow on successful E2E tests and then remove this job completely from E2E tests? https://github.com/microsoft/AL-Go/actions/workflows/CleanupTempRepos.yaml